src/os/bluestore/BlueFS.cc

```diff
@@ -1881,8 +1881,9 @@ int BlueFS::_read_random(
   FileReader *h,         ///< [in] read from here
   uint64_t off,          ///< [in] offset
   uint64_t len,          ///< [in] this many bytes
-  char *out)             ///< [out] optional: or copy it here
+  char *out)
 {
+  ceph_assert(out);
```

Comment:
i'd avoid putting `ceph_assert()` in the critical path. moreover, probably we can remove the check for null couple lines below:
```c++
      if (out) {
        // NOTE: h->bl is normally a contiguous buffer so c_str() is free.
        memcpy(out, buf->bl.c_str() + off - buf->bl_off, r);
        out += r;
      }
```